clusterDeployer.py

```diff
@@ -103,16 +103,26 @@ def _all_hosts_with_workers(self) -> Set[ClusterHost]:
     """
 
     def remove_master_dhcp_entries(self, hosts: List[ClusterHost]) -> None:
+        removed_macs = []  # type: List[str]
         for h in hosts:
-            h.prepare_remove_master_dhcp_entries(self._local_host.bridge)
-        # Restart libvirtd only once after removing all un-needed entries.
+            h.prepare_remove_master_dhcp_entries(self._local_host.bridge, removed_macs)
+        # Restart libvirtd to apply changes. Restart only once after removing all un-needed entries.
+        lh = host.LocalHost()
```

Comment:
If we really want to do this here we should at least use `self._local_host.hostconn` instead of creating a new one.

Reply:
`host.LocalHost()` is just a small object, that allows for `lsh.run()` (and similar operations). It mainly exists to provide a compatible API with `host.RemoteHost().run()`, and functions can transparently operate remote or local. 

Granted, if you just do
```python
    lh = host.LocalHost()
    lh.run("...")
```
you may also replace that with `subprocess.run()`. But `host.LocalHost()` seems just fine (at least, it does some logging).


My point is, while `host.RemoteHost()` and the additional `ssh_connect()` may be a bit expensive and avoided multiple times, there is no problem creating a lot of `host.LocalHost()` instances and use them just once.

---

Note that `host.Host.__new__()` interns the request. So `host.LocalHost()` does not even create a new instance but re-uses an existing one. But that seems a bit of an anti feature to me and something that maybe should not be relied on.

Reply:
My problem was more with the logical encapsulation of concerns.  I don't think the cluster deployer should restart libvirtd, I think it should be the responsibility of the ClusterHost (or VirtualBridge) classes.

Reply:
I see. Sorry for the lengthy unrelated reply.

Reply:
Thanks dceara and thom311. Same as above: agreed, there is no need to create new connections.